### ISSUE TITLE
Segfault in bun

### ISSUE THREAD

**Issue by alexandernst** [oven-sh/bun#15311]
### Relevant log output

```shell
Bun v1.1.36 (ededc168) Linux x64
Linux Kernel v5.4.0 | glibc v2.39
CPU: sse42 popcnt avx avx2
Args: "bun" "run" "update_assets_descriptor.js" "--file" "new/generic/intellisense/http_response.json" "--asset-id" "generic/intellisense/http_response" "--used-by" "SPA" "--assets-descriptor" "assets.json"
Features: jsc 
Builtins: "bun:main" "node:buffer" "node:crypto" "node:fs" "node:path" "node:string_decoder" "node:util" "node:util/types" 
Elapsed: 40ms | User: 34ms | Sys: 8ms
RSS: 1.07GB | Peak: 50.80MB | Commit: 1.07GB | Faults: 0
panic(main thread): Segmentation fault at address 0x6024C
oh no: Bun has crashed. This indicates a bug in Bun, not your code.
To send a redacted crash report to Bun's team,
please file a GitHub issue using the link below:
 https://bun.report/1.1.36/lr1ededc16AgggEuiw8jE+xpRu7ll8Cmw8k2C+mj9gDysm+gDk449gDwj8k3CmymtgDwws2+CA2A4khY
./generate.sh: line 35:  6209 Illegal instruction     (core dumped) bun run update_assets_descriptor.js --file new/generic/intellisense/http_response.json --asset-id generic/intellisense/http_response --used-by SPA --assets-descriptor assets.json
```

### Stack Trace (bun.report)

Bun v1.1.36 ([`ededc16`](<https://github.com/oven-sh/bun/tree/ededc168cf07afcdbe5ca34495c58df435ea526d>)) on linux x86_64 [RunCommand]

Segmentation fault at address 0x0006024C

- [`dependency.zig:851`](<https://github.com/oven-sh/bun/blob/ededc168cf07afcdbe5ca34495c58df435ea526d/src/install/dependency.zig#L851>): `src.resolver.resolver.Resolver.parsePackageJSON__anon_96842`
- [`resolver.zig:3987`](<https://github.com/oven-sh/bun/blob/ededc168cf07afcdbe5ca34495c58df435ea526d/src/resolver/resolver.zig#L3987>): `src.resolver.resolver.Resolver.dirInfoUncached`
- [`resolver.zig:2810`](<https://github.com/oven-sh/bun/blob/ededc168cf07afcdbe5ca34495c58df435ea526d/src/resolver/resolver.zig#L2810>): `src.resolver.resolver.Resolver.dirInfoCached`
- [`resolver.zig:1675`](<https://github.com/oven-sh/bun/blob/ededc168cf07afcdbe5ca34495c58df435ea526d/src/resolver/resolver.zig#L1675>): `src.resolver.resolver.Resolver.loadNodeModules`
- [`resolver.zig:2387`](<https://github.com/oven-sh/bun/blob/ededc168cf07afcdbe5ca34495c58df435ea526d/src/resolver/resolver.zig#L2387>): `src.resolver.resolver.Resolver.resolveWithoutRemapping`
- [`resolver.zig:1379`](<https://github.com/oven-sh/bun/blob/ededc168cf07afcdbe5ca34495c58df435ea526d/src/resolver/resolver.zig#L1379>): `src.resolver.resolver.Resolver.resolveAndAutoInstall`
- [`javascript.zig:2331`](<https://github.com/oven-sh/bun/blob/ededc168cf07afcdbe5ca34495c58df435ea526d/src/bun.js/javascript.zig#L2331>): `src.bun.js.javascript.VirtualMachine.resolveMaybeNeedsTrailingSlash__anon_116060`
- [`javascript.zig:2407`](<https://github.com/oven-sh/bun/blob/ededc168cf07afcdbe5ca34495c58df435ea526d/src/bun.js/javascript.zig#L2407>): `src.bun.js.api.BunObject.doResolveWithArgs__anon_109308`

Features: jsc

**Comment by Jarred-Sumner** [comment]
Fixed in Bun v1.1.37. try running `bun upgrade --canary` or wait till tomorrow for the release

This was a regression in v1.1.35

**Comment by alexandernst** [comment]
Thank you Jarred!
